client/client.go

```diff
@@ -1225,32 +1241,27 @@ func (c *Client) updateNodeFromFingerprint(response *fingerprint.FingerprintResp
 	return c.configCopy.Node
 }
 
-// resourcesAreEqual is a temporary function to compare whether resources are
-// equal. We can use this until we change fingerprinters to set pointers on a
-// return type.
-func resourcesAreEqual(first, second *structs.Resources) bool {
-	if first.CPU != second.CPU {
-		return false
-	}
-	if first.MemoryMB != second.MemoryMB {
-		return false
-	}
-	if first.DiskMB != second.DiskMB {
-		return false
-	}
-	if len(first.Networks) != len(second.Networks) {
-		return false
-	}
-	for i, e := range first.Networks {
-		if len(second.Networks) < i {
-			return false
+// updateNetworks preserves manually configured network options, but
+// applies fingerprint updates
+func updateNetworks(ns structs.Networks, up structs.Networks, c *config.Config) structs.Networks {
+	if c.NetworkInterface == "" {
+		ns = up
+	} else {
+		// if a network is configured, use only that network
+		// use the fingerprinted data
+		for _, n := range up {
+			if c.NetworkInterface == n.Device {
+				ns = []*structs.NetworkResource{n}
+			}
 		}
-		f := second.Networks[i]
-		if !e.Equals(f) {
-			return false
+		// if not matched, ns has the old data
+	}
```

Comment:
Our style is to add a newline after most blocks.
```suggestion
	}

```